### DISCUSSION TITLE
Fatal error while launching run_and_log

### DISCUSSION THREAD

**Discussion by jacobinho86** [kaiwan/seals#11] [Q&A] [answered]
I am trying to setup an emulator for the rpi board, because I want to try out the new kernel module that I built for it, but I get the following fatal error when running the script run_and_log:

![image](https://github.com/kaiwan/seals/assets/85832964/26e719e8-2106-4792-a0d7-e4def1505f37)

I am lost here, I have installed the toolchain, so any insight you can share about this problem will be very much appreciated.

Kind Regards

**Comment by kaiwan** [comment]
Hi @jacobinho86 , thanks for raising this..
As you've installed a toolchain (it should be an x86_64-to-AArch64 one), the likely issue is that it's location isn't setup in your `PATH` env variable and is hence not being located by the SEALS build script.. 

Pl try and update PATH to include the toolchain location on disk. Test by running on the shell, for example, 
`aarch64-none-linux-gnu-gcc -v`
It working implies the PATH's set correctly; now retry the `run_and_log` script.
Do let me know..

**Reply by jacobinho86** [reply]
Thanks for your answer Kaiwan, I've checked my set-up and this is what I gathered:

1. This is the gcc for the toolchain that I installed (maybe is the wrong one?):

`c2kp@osboxes:~$ aarch64-linux-gnu-gcc -v`
`Using built-in specs.`
`COLLECT_GCC=aarch64-linux-gnu-gcc`
`COLLECT_LTO_WRAPPER=/usr/lib/gcc-cross/aarch64-linux-gnu/11/lto-wrapper`
`Target: aarch64-linux-gnu`

2. This is where it is installed:

`c2kp@osboxes:~$ which aarch64-linux-gnu-gcc`
`**/usr/bin/**aarch64-linux-gnu-gcc`

3. And the root path is in the `$PATH` variable

`c2kp@osboxes:~$ echo $PATH`
`/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin`

This is the toolchain that I used to built the kernel module, but maybe, this is not the correct toolchain to use in SEALS (not supported?). 

Regards

**Comment by kaiwan** [comment]
Okay.. toolchain seems ok, at first glance at least.
Have you updated the CXX variable in the build.config file?
(the script looks up ${CXX}gcc )

**Reply by jacobinho86** [reply]
Thanks again Kaiwan!, setting the variable was my missing step, which I've done it:

`2kp@osboxes:~/Documents/seals$ cat build.config | grep CXX`
`#CXX=arm-none-linux-gnueabihf- # toolchain prefix backup!!!!`
**`CXX=aarch64-linux-gnu- # toolchain prefix`**

Now I get anther error:

![unnamed](https://github.com/kaiwan/seals/assets/85832964/81032978-7005-4c83-a81f-e53e1100b75a)

But I'll follow the recommendation, fingers crossed hopping that this is it.  

Kind Regards

**Reply by kaiwan** [reply, marked as answer]
Great. Pl follow it... also, the error msg shows that the yad utility needs to be installed..

**Reply by jacobinho86** [reply]
It worked! Thanks for the support :)

**Reply by kaiwan** [reply]
Welcome!
